apis/cloudidentity/v1beta1/membership_types.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	commonv1alpha1 "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/common/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var CloudIdentityMembershipGVK = GroupVersion.WithKind("CloudIdentityMembership")
+
+// CloudIdentityMembershipSpec defines the desired state of CloudIdentityMembership
+// +kcc:proto=mockgcp.cloud.cloudidentity.groups.v1beta1.Membership
+type CloudIdentityMembershipSpec struct {
+	/* Immutable. Optional. The service-generated name of the resource. Used for acquisition only. Leave unset to create a new resource. */
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Immutable.
+	// +required
+	GroupRef refsv1beta1.GroupRef `json:"groupRef"`
+
+	// Immutable. The `EntityKey` of the member. Either `member_key` or `preferred_member_key` must be set when calling MembershipsService.CreateMembership but not both; both shall be set when returned.
+	MemberKey *MembershipMemberKey `json:"memberKey,omitempty"`
+
+	// Immutable. Required. Immutable. The `EntityKey` of the member.
+	// +required
+	PreferredMemberKey MembershipPreferredMemberKey `json:"preferredMemberKey"`
```

Comment:
nit: when copying and pasting code from the "types.generated.go" file, we should keep the `// +kcc:proto:field` annotations. While they don't affect the final CRD rendering, they are useful for our internal tooling.

Reply:
It wasnt in the originally generated file : https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/422c59d957ec31e714681ec9c9f5c965ad3952ac/apis/cloudidentity/v1beta1/membership_types.go